crates/expressions/src/lib.rs

```diff
@@ -19,6 +19,14 @@ pub struct Resolver {
     providers: Vec<Box<dyn Provider>>,
 }
 
+#[derive(Default)]
+pub struct PreparedResolver {
```

Comment:
Nit-shedding (bike-knitting?): maybe `ResolvedVariables`?

Reply:
I prefer a `Resolver`-based name because the purpose of it is to resolve templates; that it holds the resolved variables is an implementation detail.